ipamd/ipamd.go

```diff
@@ -348,177 +352,254 @@ func (c *IPAMContext) StartNodeIPPoolManager() {
 }
 
 func (c *IPAMContext) updateIPPoolIfRequired() {
-	c.retryAllocENIIP()
```

Comment:
This logic was moved into increaseIPPool, since that's what its doing, and if WARM_IP_TARGET is set then its not always a retry.